pkg/apis/sources/v1beta1/ping_lifecycle.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+)
+
+const (
+	// PingSourceConditionReady has status True when the PingSource is ready to send events.
+	PingSourceConditionReady = apis.ConditionReady
+
+	// PingSourceConditionSinkProvided has status True when the PingSource has been configured with a sink target.
+	PingSourceConditionSinkProvided apis.ConditionType = "SinkProvided"
+
+	// PingSourceConditionDeployed has status True when the PingSource has had it's receive adapter deployment created.
+	PingSourceConditionDeployed apis.ConditionType = "Deployed"
+)
```

Comment:
I see more conditions in v1alpha2 type https://github.com/knative/eventing/blob/82399b82ed810063f0389e867e6f00de5c13a4a8/pkg/apis/sources/v1alpha2/ping_lifecycle.go#L33

Why do we reduce the the number of them?

Reply:
`ValidSchedule` is not needed anymore since the validation is done in the webhook. `ResourcesCorrect` has long been deprecated. And `EventTypeProvided` is not used.